src/core/meshcop/commissioner.cpp

```diff
@@ -800,26 +809,20 @@ void Commissioner::HandleLeaderKeepAliveResponse(Coap::Message *         aMessag
 
     StateTlv state;
 
-    VerifyOrExit(mState == OT_COMMISSIONER_STATE_ACTIVE, SetState(OT_COMMISSIONER_STATE_DISABLED));
-    VerifyOrExit(aResult == OT_ERROR_NONE && aMessage->GetCode() == OT_COAP_CODE_CHANGED,
-                 SetState(OT_COMMISSIONER_STATE_DISABLED));
+    VerifyOrExit(mState == OT_COMMISSIONER_STATE_ACTIVE, Stop(false));
```

Comment:
No need to stop the commissioner. Maybe ignore the message.

Reply:
The current code will cause a petitioning commissioner to stop if KeepAlive response is received. 
 I think the chance to happen is negligible.

Reply:
yes, this will rarely happen if all things are going well. But, we would say, "stop the commissioner when receiving an unexpected message" is wrong. Agree?